web/modules/custom/parser/src/Form/FilesAndTablesManagerForm.php

```diff
@@ -0,0 +1,386 @@
+<?php
+
+namespace Drupal\parser\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\parser\Reader\CsvReader;
+use Drupal\parser\Reader\DiscountReader;
+use Drupal\parser\Reader\ExcelReader;
+use Drupal\parser\Reader\YamlReader;
+use Drupal\parser\Writer\DB\EntitlementsWriter;
+use Drupal\parser\Writer\DB\Rates400NGWriter;
+use Drupal\parser\Writer\DB\CsvWriter;
+use Drupal\parser\Writer\DB\DiscountWriter;
+use Drupal\parser\Writer\DB\ZipCodesWriter;
+use Drupal\Core\Url;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+
+/**
+ * Class FilesAndTablesManagerForm.
+ */
+class FilesAndTablesManagerForm extends ConfigFormBase {
+
+  /**
+   * Variable containing the database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $db;
+
+  /**
+   * Variable containing the entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entity;
+
+  /**
+   * FilesAndTablesManagerForm constructor.
+   *
+   * Needed for dependency injection.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, Connection $db, EntityTypeManager $entity) {
+    parent::__construct($config_factory);
+    $this->db = $db;
+    $this->entity = $entity;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('database'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'parser.filesandtablesmanager',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'files_and_tables_manager_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['#tree'] = TRUE;
+
+    $form['zip_3'] = [
+      '#type' => 'details',
+      '#title' => $this->t('Zip 3'),
+    ];
+
+    $form['zip_3']['file'] = [
+      '#type' => 'managed_file',
+      '#title' => $this->t('Zip_3'),
+      '#upload_validators' => [
+        'file_validate_extensions' => ['csv'],
+      ],
+    ];
+
+    $form['zip_3']['truncate'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Clean table (removes all data previously stored)'),
+    ];
+
+    $form['zip_3']['link'] = [
+      '#title' => $this->t('What data is in this table?'),
+      '#type' => 'link',
+      '#url' => Url::fromRoute('<front>'),
+    ];
```

Comment:
this could be changed to $form['zip_3'] = $this->buildZip3Wrapper(); 
Or something like that.
And the same thing for the cases below.

Reply:
As per discussion with Gerardo and Davey, this is how Drupal developers normally write/read the creation of a form.